libraries/aws_guardduty_detector.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require 'aws_backend'
+
+class AwsGuardDutyDetector < AwsResourceBase
+  name 'aws_guardduty_detector'
+  desc 'Verifies settings for a Detector'
+  example "
+    describe aws_guardduty_detector(detector_id: 'detector-id') do
+      it { should exist }
+    end
+  "
+
+  def initialize(opts = {})
+    opts = { detector_id: opts } if opts.is_a?(String)
+    super(opts)
+    validate_parameters(required: [:detector_id])
+    @display_name = opts[:detector_id]
+
+    catch_aws_errors do
+      resp = @aws.guardduty_client.get_detector({ detector_id: opts[:detector_id] })
```

Comment:
A simple validation of the user-provided `detector_id` before sending it to the API would be a safer operation.
https://docs.aws.amazon.com/guardduty/latest/APIReference/API_GetDetector.html#API_GetDetector_RequestSyntax

Reply:
Added a between 1 and 300 chars check in latest commit.